src/django_components/templatetags/component_tags.py

```diff
@@ -427,6 +429,20 @@ def fill(parser: Parser, token: Token, tag_spec: TagSpec) -> FillNode:
       {% endfill %}
     {% endcomponent %}
     ```
+
+    ### Accessing slot data and default content on the default slot
+
+    To access slot data and the default slot content on the default slot,
+    use `{% fill %}` without the `name` kwarg:
+
+    ```django
+    {% component "button" %}
+      {% fill data="slot_data" default="default_slot" %}
```

Comment:
For django-vue, beside the "passthrough slots" feature, I also needed a way to use the `data` and `default` kwargs on a default fill.

E.g. in Vue you can do `#slot_name="{ item }"` to access the slot data of the slot `slot_name`

  ```vue
  <MyTable>
      <template #default="{ item }">
          {{ item }}
      </template>
  </MyTable>
  ```

Because in Vue, when you pass in a default fill, it's as if the fill was assigned to a fill named "default".

So in Vue, these two are identical:

  ```vue
  <MyTable>
          {{ item }}
  </MyTable>

  <MyTable>
      <template #default>
          {{ item }}
      </template>
  </MyTable>
  ```

---

Now, in my head, the point of the default fill is that you don't know / care what is the slot's name. So my initial approach for accessing slot data of a default fill in django-components was to simply omit the `name=...` kwarg:

```django
{% component "button" %}
  {% fill data="slot_data" default="default_slot" %}
    You clicked me {{ slot_data.count }} times!
    {{ default_slot }}
  {% endfill %}
{% endcomponent %}
```

However, later, as I had to redo the slot implementation, I realized that omitting the name would not work nicely with the "passthrough slots" feature. Because the default slot was stored under an internal key. So we'd either allow something like this to target the default slot:
```django
{% fill name="_DJANGO_COMPONENTS_DEFAULT_SLOT" %}
```
Or for the passthrough slots, there'd have to be an extra if / else:
```django
{% component "slotted" %}
    {% for slot_name in slots %}
      {% if slot_name == "_DJANGO_COMPONENTS_DEFAULT_SLOT" %}
        {% fill %}
            {% slot name=slot_name default / %}
        {% endfill %}
      {% else %}
        {% fill name=slot_name %}
            {% slot name=slot_name / %}
        {% endfill %}
      {% endif %}
    {% endfor %}
{% endcomponent %}
```

So in the end I decided to go with Vue's implementation, which is that:
1. Default fill is assigned to a fill named "default"
2. If a slot is marked as default, it will use the "default" fill.

---

In the end, I kept both approaches (omitting fill's name and using "default" fill, which means that fills can now be written in 3 ways:

1. Fill with regular name
  ```django
  {# This will target the `header` slot, irrespective of whether it's default #}
  {% fill name="header" / %}
  ```
2. Fill with name "default"
```django
  {# This will target the slot with the `default` flag #}
  {% fill name="default" / %}
  ```
3. Fill without a name
```django
  {# This will target the slot with the `default` flag #}
  {% fill / %}
  ```
---

Note, though, that our implementation differs from Vue's. Because in Vue, the default slot is defined by omitting the name:

```vue
<slot />
```

In which case Vue gives this slot a name `default`.

However, our case, we use the `default` flag, but still require some name:

```django
{% slot "my_slot" default / %}
```

Which means that a single slot may be targetted in two ways:

1. By the slot name 
2. As the "default" slot.

E.g. in the case below, both fills target the same slot.

```django
{% component 'test' %}
    {% fill %} ABC {% endfill %}
    {% fill "my_slot" %} XYZ {% endfill %}
{% endcomponent %}
```

In such cases, we raise an error. (See tests like this one https://github.com/EmilStenstrom/django-components/pull/758/files#diff-ea7f91c03e3b214aa30b65d58c3ea193c51868981b298410097f11e6f457fa01R366)

Reply:
UPDATE: So I reverted the "`{% fill %}` without a name is a default slot" feature in https://github.com/EmilStenstrom/django-components/pull/758/commits/072c4f45659fecdab228f99c43dd540dba0fa22c. It was just simpler to document the changes while always using `name="default"`.